File: python/InsertionSort.py
# 問題1: Insertion Sort を使って、整数配列 [12, 11, 13, 5, 6] をソートしてください。


def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
    return arr

# テスト
arr = [9, 5, 1, 4, 3]
print("ソート前:", arr)
print("ソート後:", insertion_sort(arr))

File: c++/InsertionSort.cpp
// 問題1: Insertion Sort を使って、整数配列 [12, 11, 13, 5, 6] をソートしてください。


#include <iostream>
#include <vector>
using namespace std;

void insertionSort(vector<int>& arr) {
    for (int i = 1; i < arr.size(); i++) {
        int key = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

int main(){
    vector<int> arr = {12, 11, 13, 5, 6};
    cout << "Before sorting: ";
    for (int num : arr)
        cout << num << " ";
    cout << "\n";
    
    insertionSort(arr);
    
    cout << "After sorting: ";
    for (int num : arr)
        cout << num << " ";
    cout << "\n";
    return 0;
}
